pkg/openstack/instances.go

```diff
@@ -633,6 +633,44 @@ func nodeAddresses(srv *servers.Server, ports []ports.Port, networkingOpts Netwo
 		return nil, err
 	}
 
+	// Add the addresses assigned on subports via trunk
+	// This exposes the vlan networks to which subports are attached
+	subportAddresses := make(map[string][]Address)
+
+	for _, port := range ports {
+		for _, subport := range port.TrunkDetails.SubPorts {
+			p, err := neutronports.Get(client, subport.PortID).Extract()
+			if err != nil {
+				klog.Errorf("Failed to get subport %s details: %v", subport.PortID, err)
+				continue
+			}
+			n, err := networks.Get(client, p.NetworkID).Extract()
+			if err != nil {
+				klog.Errorf("Failed to get subport %s network details: %v", subport.PortID, err)
+				continue
+			}
+			for _, fixedIP := range p.FixedIPs {
+				isIPv6 := net.ParseIP(fixedIP.IPAddress).To4() == nil
+				if !(isIPv6 && networkingOpts.IPv6SupportDisabled) {
+					addr := Address{IPType: "fixed", Addr: fixedIP.IPAddress}
+					subportAddresses[n.Name] = append(subportAddresses[n.Name], addr)
+					klog.V(5).Infof("Node '%s' is added subport '%s' address '%s/%s'", srv.Name, p.Name, n.Name, addr)
+				}
+			}
+		}
+	}
+
+	for nw, ips := range subportAddresses {
+		srvAddresses, ok := addresses[nw]
+		if !ok {
+			addresses[nw] = ips
+		} else {
+			// this is to take care the corner case
+			// where the same network is attached to the node both directly and via trunk
+			addresses[nw] = append(srvAddresses, ips...)
+		}
+	}
+
```

Comment:
Suggestion: we're already looping over this above (in the FixedIPs loop). You could move this to there and eliminate subportAddresses.

Reply:
will check